src/Geta.Optimizely.Sitemaps/Configuration/SitemapOptions.cs

```diff
@@ -1,9 +1,22 @@
-﻿namespace Geta.Optimizely.Sitemaps.Configuration
+using System.Linq;
+using Geta.Optimizely.Sitemaps.Services;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Geta.Optimizely.Sitemaps.Configuration
 {
     public class SitemapOptions
     {
         public bool EnableRealtimeSitemap { get; set; } = false;
         public bool EnableRealtimeCaching { get; set; } = true;
         public bool EnableLanguageDropDownInAdmin { get; set; } = false;
+
+        public void SetAugmenterService<T>(IServiceCollection services) where T : class, IUriAugmenterService
+        {
+            var augmenterService = services.First(sd => sd.ServiceType == typeof(IUriAugmenterService));
```

Comment:
`SitemapOptions` is just a DTO, it should not register services by itself. It should just store the type of augmented service to register and then `ServiceCollectionExtensions` will register it.
See how it is done in ProductFeed:
https://github.com/Geta/geta-optimizely-productfeed/blob/master/src/Geta.Optimizely.ProductFeed/Configuration/ProductFeedOptions.cs#L47
```
     public Type Filter { get; set; }

     public ProductFeedOptions<TEntity> SetFilter<TFilter>() where TFilter : IProductFeedFilter<TEntity>
      {
          Filter = typeof(TFilter);
          return this;
      }
```